### PR TITLE
[SPARK-43125][CONNECT] Fix Connect Server Can't Handle Exception With Null Message

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
Fix the bug when Connect Server throw Exception without message.

### Why are the changes needed?
Fix bug

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Unnecessary

### CODE REVIEW

**Comment by Hisoka-X** [comment]
@martin-g @HyukjinKwon PTAL. Thanks

**Comment by LuciferYang** [comment]
Could you add a test case for this scenario?

**Comment by LuciferYang** [comment]
hmm.. there is no `[SERVER]` tag, we should remove if from pr title

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala [hunk 1]
```diff
@@ -85,7 +86,7 @@ class SparkConnectService(debug: Boolean)
             .setDomain("org.apache.spark")
             .putMetadata("classes", compact(render(allClasses(st.getClass).map(_.getName))))
             .build()))
-      .setMessage(StringUtils.abbreviate(st.getMessage, 2048))
+      .setMessage(if (message != null) message else "")
```
Comment (amaliujia) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala:89]: I am wondering when `message` could be `null` than an empty string? If this is a valid case we can use `st: Optiona[Throwable]` instead?

Comment (Hisoka-X) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala:89]: > I am wondering when message could be null than an empty string?

`NullPointerException`

> If this is a valid case we can use st: Optiona[Throwable] instead?

Can't, like NPE, the `Optiona[Throwable]` would not empty, but message in NPT still `null`.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Comment by WweiL** [comment]
@Hisoka-X @HyukjinKwon Any reason we drop the `abbreviate`?

**Comment by Hisoka-X** [comment]
> @Hisoka-X @HyukjinKwon Any reason we drop the `abbreviate`?

Sorry, I don't get it. What do you mean about `drop the abbreviate`? This PR just do null check.

**Comment by WweiL** [comment]
> > @Hisoka-X @HyukjinKwon Any reason we drop the `abbreviate`?
> 
> Sorry, I don't get it. What do you mean about `drop the abbreviate`? This PR just do null check.

Ah sorry! I didn't see you did the abbreviate in the top, please ignore this!

**Comment by WweiL** [comment]
> > @Hisoka-X @HyukjinKwon Any reason we drop the `abbreviate`?
> 
> Sorry, I don't get it. What do you mean about `drop the abbreviate`? This PR just do null check.

Ah sorry! I didn't see you did the abbreviate in the top, please ignore this!

**Comment by WweiL** [comment]
@Hisoka-X I think we also change this line?
https://github.com/apache/spark/pull/40780/files#diff-d21a96890e660df398527183a191b0d0c73521aada856e2491c33243f269fdceR128

I could add it to my ongoing PR https://github.com/apache/spark/pull/40785

**Comment by Hisoka-X** [comment]
> @Hisoka-X I think we also change this line? https://github.com/apache/spark/pull/40780/files#diff-d21a96890e660df398527183a191b0d0c73521aada856e2491c33243f269fdceR128
> 
> I could add it to my ongoing PR #40785

OK for me, but I just want to remind, add null check just want to avoid throw NPE in `setMessage`, not want to change `null` to  `""`

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala (hunk 1 above)
Comment (allisonwang-db) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala:89]: We should still abbreviate this message when it's not null

**Comment by WweiL** [comment]
> > @Hisoka-X I think we also change this line? https://github.com/apache/spark/pull/40780/files#diff-d21a96890e660df398527183a191b0d0c73521aada856e2491c33243f269fdceR128
> > I could add it to my ongoing PR #40785
> 
> OK for me, but I just want to remind, add null check just want to avoid throw NPE in `setMessage`, not want to change `null` to `""`

I see. With additional look I think here it's better to keep what it is. as Status's description is nullable:

``` private Status(Code code, @Nullable String description, @Nullable Throwable cause) {
    this.code = checkNotNull(code, "code");
    this.description = description;
    this.cause = cause;
  }

public Status withDescription(String description) {
    if (Objects.equal(this.description, description)) {
      return this;
    }
    return new Status(this.code, description, this.cause);
  }

```

**Pull request closed without merging**